chain/mining.go

```diff
@@ -108,16 +108,17 @@ func (bm *BuiltinMining) BuildBlock(height uint32, chordID []byte, winnerHash co
 	if err != nil {
 		return nil, err
 	}
+	consensusData := rand.Uint64()
 	curBlockHash := DefaultLedger.Store.GetCurrentBlockHash()
-	curStateHash := GenerateStateRoot(txnList)
+	curStateHash := GenerateStateRoot(txnList, height, consensusData)
 	header := &Header{
 		BlockHeader: BlockHeader{
 			UnsignedHeader: &UnsignedHeader{
 				Version:          HeaderVersion,
 				PrevBlockHash:    curBlockHash.ToArray(),
 				Timestamp:        timestamp,
 				Height:           height,
-				ConsensusData:    rand.Uint64(),
+				ConsensusData:    consensusData,
```

Comment:
My suggestion:
1. Rename ConsensusData to RandomBeacon
2. Change it to 64 bytes (to provide enough entropy)
3. Do not use math.rand (it's not cryptographically secure), use RandomBytes defined in crypto/util/util.go

Reply:
@yilunzhang  Just only **random from proposer** for ConsensusData is not enough. 

As an ID register who aim to gain appointed chord position, its pubKey and expected ID is determinate. The hash of generateID txn is also determinate too.

The position picker can exhaust applicable `ConsensusData` **in advance** and waiting for its conspirator win a proposal right.

So, We should add `additional constraint`(**factor from out of current proposer**) for ConsensusData/RandomBeacon generation.


> 2. Change it to 64 bytes to provide enough entropy for most purposes

Promote Uint64 to 64Bytes is expensive for size of Blk.Header.
In v0.9, the average of Blk.Size with only 1 transaction(coinbase) is around 390+ bytes.  +56 bytes means that Blk.Header increase 15% size.

Reply:
For 1: You are right, I forget this one. We need to use
id = hash(current block random beacon, prev block hash, txn hash)
@NoelBright 

For 2: I didn't realize our current block size is so small. What about 32 bytes then?

Reply:
ok